src/mikeio/lazy.py

```diff
@@ -0,0 +1,1168 @@
+                        continue
+                else:
+                    timestep_out = timestep_in - start_step
+
+                # Compute rolling statistic for each item
+                for item_idx in range(n_items):
+                    # Get the window data
+                    if timesteps_filled < window:
+                        # Partial window at the start
+                        window_data = buffers[item_idx][:timesteps_filled]
+                    else:
+                        # Full window - need to handle circular buffer
+                        window_data = buffers[item_idx]
+
+                    # Apply statistic along time axis (axis=0)
+                    result = np.apply_along_axis(stat_func, 0, window_data)
+
+                    # Replace NaN with delete value
+                    result[np.isnan(result)] = context.deletevalue
+
+                    # Write to output (use WriteItemTimeStepNext for sequential writing)
+                    itemdata = context.dfs_in.ReadItemTimeStep(
+                        context.item_numbers[item_idx] + 1, timestep_in
+                    )
+                    context.dfs_out.WriteItemTimeStepNext(itemdata.Time, result)
+
+            buffer_idx += 1
+
+    def _process_with_aggregate(
+        self, context: ExecutionContext, aggregate_op: AggregateOp
+    ) -> None:
+        """Process data with temporal aggregation."""
+        stat_func = aggregate_op.get_stat_func()
+        time_info = context.time_info
+
+        # Determine time range
+        if time_info is not None:
+            start_step = time_info.start_step
+            end_step = time_info.end_step
+        else:
+            start_step = 0
+            end_step = context.dfs_in.FileInfo.TimeAxis.NumberOfTimeSteps
+
+        # Check if this is a layered dfsu (Z coordinate at index 0)
+        is_layered = (
+            len(context.item_numbers) > 0
+            and context.dfs_in.ItemInfo[context.item_numbers[0]].Name == "Z coordinate"
+        )
+
+        # Handle Z coordinate for layered dfsu
+        if is_layered:
+            # Write Z coordinate (first item, first timestep, no aggregation)
+            zn_itemdata = context.dfs_in.ReadItemTimeStep(1, start_step)
+            context.dfs_out.WriteItemTimeStepNext(0.0, zn_itemdata.Data)
+            # Process remaining items (skip Z coordinate)
+            item_numbers_to_aggregate = context.item_numbers[1:]
+        else:
+            item_numbers_to_aggregate = context.item_numbers
+
+        n_items = len(item_numbers_to_aggregate)
+
+        if n_items == 0:
+            return
+
+        # Get data shape from first item to aggregate
+        first_itemdata = context.dfs_in.ReadItemTimeStep(
+            item_numbers_to_aggregate[0] + 1, start_step
+        )
+        n_elements = len(first_itemdata.Data)
+        n_timesteps = end_step - start_step
+
+        # Create buffers: [n_items][n_timesteps, n_elements]
```

Comment:
Allocating these arrays defies the entire purpose of the `lazy` module (files larger than memory). This has to be done in a different way. 🤬